src/NumericalAlgorithms/SphericalHarmonics/WignerThreeJ.cpp

```diff
@@ -0,0 +1,70 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "NumericalAlgorithms/SphericalHarmonics/WignerThreeJ.hpp"
+
+#include <limits>
+
+#include "Utilities/ErrorHandling/Error.hpp"
+#include "Utilities/ErrorHandling/ExpectsAndEnsures.hpp"
+
+extern "C" {
+void drc3jj_(const double& l2, const double& l3, const double& m2,
+             const double& m3, double& l1min, double& l1max, double* answer,
+             const int& answersize, int& error_code, const double& doublemax);
+}
+
+WignerThreeJ::WignerThreeJ(const size_t l2, const int m2, const size_t l3,
+                           const int m3)
+    : l2_(double(l2)),
+      m2_(m2),
+      l3_(double(l3)),
+      m3_(m3),
+      l1_min_(static_cast<size_t>(
+          std::max(std::abs(static_cast<int>(l2) - static_cast<int>(l3)),
+                   std::abs(m2 + m3)))),
+      l1_max_(l2 + l3),
+      up_to_date_(false),
+      coefs_(l1_max_ - l1_min_ + 1) {
+  if (UNLIKELY(std::abs(m2) > static_cast<int>(l2))) {
+    ERROR("WignerThreeJ: m2 is " << m2 << " but l2 is " << l2);
+  }
+  if (UNLIKELY(std::abs(m3) > static_cast<int>(l3))) {
+    ERROR("WignerThreeJ: m3 is " << m3 << " but l3 is " << l3);
```

Comment:
These should also say something like "abs(m2) must be larger than l2" so the user knows what to change if they get this error